src/django_pg_bulk_update/query.py

```diff
@@ -220,12 +227,16 @@ def _validate_set_functions(model, fds, functions):
 
     for f in fds:
         field = f.get_field(model)
-        if getattr(field, 'auto_now', False):
-            f.set_function = NowSetFunction(if_null=False)
-        elif getattr(field, 'auto_now_add', False):
-            f.set_function = NowSetFunction(if_null=True)
-        else:
-            f.set_function = functions.get(f.name)
+
+        set_func = functions.get(f.name)
+        if set_func is None:
+            # Note that we should always respect any user provided set functions
+
+            if getattr(field, 'auto_now', False):
+                f.set_function = NowSetFunction(if_null=False)
+            elif getattr(field, 'auto_now_add', False):
+                f.set_function = NowSetFunction(if_null=True)
+        f.set_function = set_func
```

Comment:
1. There is a bug: you always reset `f.set_function` to `set_func` at the end, even if it is `None` and has been set to `NowSetFunction(...)`
2. I'd rather rewrite it like:
```suggestion
        # We should always respect any user provided set functions
        set_func = functions.get(f.name)
        if set_func is None:
            if getattr(field, 'auto_now', False):
                set_func = NowSetFunction(if_null=False)
            elif getattr(field, 'auto_now_add', False):
                set_func = NowSetFunction(if_null=True)
                
        f.set_function = set_func
```